test/e2e/tests/test_usergroup.py

```diff
@@ -0,0 +1,93 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the Elasticache UserGroup resource
+"""
+import pytest
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+
+from e2e import service_marker, CRD_GROUP, CRD_VERSION
+from e2e.bootstrap_resources import get_bootstrap_resources
+
+RESOURCE_PLURAL = "usergroups"
+KIND_NAME = "UserGroup"
+
+
+@pytest.fixture(scope="module")
+def bootstrap_resources():
+    return get_bootstrap_resources()
+
+
+# setup/teardown for test_user_group_create_update
+@pytest.fixture(scope="module")
+def user_group_create():
+    user_group_name = random_suffix_name("ack-usergroup", 32)
+
+    reference = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL, user_group_name, namespace="default")
+
+    user_group = {
+        "apiVersion": CRD_GROUP + "/" + CRD_VERSION,
+        "metadata": {
+            "name": reference.name
+        },
+        "kind" : KIND_NAME,
+        "spec": {
+            "engine": "redis",
+            "userGroupID": user_group_name,
+            "userIDs": [
+                "default"
+            ]
+        }
+    }
```

Comment:
This can be moved to `/test/e2e/resources` as `UserGroup.json` and referenced here using `load_elasticache_resource` from e2e.